org.eclipse.lsp4e/src/org/eclipse/lsp4e/MarkerAttributeComputer.java

```diff
@@ -57,21 +58,41 @@ public int attributeCount() {
 		return 8;
 	}
 
+	/**
+	 * Computes the attributes of a marker derived from its resource.
+	 *
+	 * @param resource,
+	 *            the {@link IResource} that contains the document.
+	 * @return a map with the marker attributes
+	 */
+	public @Nullable Map<String, Object> computeMarkerAttributes(@NonNull IResource resource) {
+		return null;

```

Comment:
I could return here as well new ArrayList<>() or Collections.emptyList() if you prefer

Reply:
Still, I really don't think this responsibility belongs to LSP4E. LSP4E is about binding LSP info to Eclipse model. What you need here (specific attributes for your specific marker type) doesn't really seem to be related to LSP at all, but more an applicative constraint on your end that depends more on your marker type than on the Language Server.